### PR TITLE
Add UNet3D convergence test on TPU v3-8

### PR DESCRIPTION
**Pull request by thisisalbertliang**
Resolves #616 

## Pull Request Summary
@thisisalbertliang developed [a UNet3D PyTorch/XLA codebase](https://github.com/thisisalbertliang/training/tree/master/image_segmentation/pytorch) that can train [MLPerf's UNet3D](https://github.com/mlcommons/training/tree/master/image_segmentation/pytorch) on both TPU and GPU.
This PR adds a convergence test for this UNet3D model on TPU v3-8

## Notes
- Currently, this convergence test still has trouble starting up on TPU v3-8 in GKE due to some configuration errors. In particular, the UNet3D training currently crashes upon start-up from an error: `tensorflow/core/framework/op_kernel.cc:1745] OP_REQUIRES failed at tpu_execute_op.cc:266 : INVALID_ARGUMENT: Run-time shape mismatch for XRTExecute argument[0] (2289320477493819). Expected element_type: F32`. The source of this error is likely unrelated to @thisisalbertliang's UNet3D PyTorch/XLA codebase.
    - For Googlers, see [b/229152397](https://b.corp.google.com/229152397) for more details on this. 
- To clarify, the code used by this convergence test is actually not the `master` branch of the UNet3D PyTorch/XLA codebase. Instead, it uses the `test-grid-integration` branch. This is because the `master` branch currently runs the model evaluation on TPU/GPU, which runs into OOM issues (described in detail in [b/220350444](https://b.corp.google.com/issues/220350444)). The `test-grid-integration` branch refactors the model evaluation to run on CPU, which has proved to successfully train the UNet3D model to convergence to `90.8%` accuracy in the past (see [b/228597462](https://b.corp.google.com/issues/228597462) for a successful convergence run on TPU v4-8)

## For Googlers
See b/229152397 for more details.

### CODE REVIEW

File: tests/pytorch/nightly/unet3d.libsonnet [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+local utils = import 'templates/utils.libsonnet';
+
+{
+  local command_common = |||
+    git clone https://github.com/thisisalbertliang/training.git -b test-grid-integration unet3d_test
+    pip3 install -r unet3d_test/image_segmentation/pytorch/requirements.txt
+    pip3 install tqdm
+
+    python3 unet3d_test/image_segmentation/pytorch/main.py --data_dir /datasets/kits19 \
+    --epochs 4000 \
```
Comment (miladm) [inline tests/pytorch/nightly/unet3d.libsonnet:27]: `300` should be sufficient.

File: tests/pytorch/nightly/unet3d.libsonnet [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    --evaluate_every 300 \
+    --start_eval_at 300 \
+    --quality_threshold 0.5 \
```
Comment (miladm) [inline tests/pytorch/nightly/unet3d.libsonnet:30]: Based on offline conversation, suggesting `0.2`

**Review by miladm — APPROVED** [review]
Config LGTM - short of passing the CI tests.

@miaoshasha please add your approval.

**Review by miladm — APPROVED** [review]

File: tests/pytorch/nightly/unet3d.libsonnet (hunk 2 above)
Comment (thisisalbertliang) [reply tests/pytorch/nightly/unet3d.libsonnet]: Sure. I have lowered the model quality threshold to `0.2`.

The convergence test will now pass if the model can reach 0.2 accuracy within 500 epochs (which allows for 2 tries at model evaluation)

File: tests/pytorch/nightly/unet3d.libsonnet (hunk 1 above)
Comment (thisisalbertliang) [reply tests/pytorch/nightly/unet3d.libsonnet]: Ok. To clarify, this `--epochs` param refers to the maximum number of epochs that the training will run for.

For safety, I have lowered `--epochs` to `501` and set `--evaluate_every` to `250` and `start_eval_at` to `250`. This will give the model two chances at passing the `0.2` accuracy test.

File: tests/pytorch/nightly/unet3d.libsonnet (hunk 2 above)
Comment (miladm) [reply tests/pytorch/nightly/unet3d.libsonnet]: This config sounds acceptable. We may end up tweaking it after the first test attempt.

**Review by miaoshasha — APPROVED** [review]
LGTM
